content/docs/rosa/alb-nlb-ossm-e2e/index.md

```diff
@@ -0,0 +1,445 @@
+We use cert-manager to issue a certificate for bookinfo application. The following step creates a self-signed certificate. if you have a certificate you can skip these steps
+
+1. Create a project (namespace) in the ROSA cluster.
+```bash
+oc new-project cert-manager --display-name="Certificate Manager" --description="Project to manage certificate LCM"
+```
+
+2. Install the cert-manager Operator
+
+```bash
+
+cat <<EOF | oc create -f -
+# apiVersion: operators.coreos.com/v1
+# kind: OperatorGroup
+# metadata:
+#   generateName: cert-manager-
+#   namespace: cert-manager
+# spec:
+#   targetNamespaces:
+#     - cert-manager-operator
+---
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: cert-manager
+  namespace: openshift-operators
+spec:
+  channel: stable
+  installPlanApproval: Automatic
+  name: cert-manager
+  source: community-operators
+  sourceNamespace: openshift-marketplace
+EOF
+```
+
+3. Create a cluster issuer
+
+```bash
+cat <<EOF | oc create -f -
+apiVersion: cert-manager.io/v1
+kind: ClusterIssuer
+metadata:
+  name: self-signed-issuer
+spec:
+  selfSigned: {}
+EOF
+```
+4. Create a self-signed certificate
+```bash
+cat <<EOF | oc create -f -
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: bookinfo-self-sgined-cert
+  namespace: ossm
+spec:
+  commonName: securedbookinfo.com
+  dnsNames:
+    - securedbookinfo.com
+  issuerRef:
+    kind: ClusterIssuer
+    name: self-signed-issuer
+  privateKey:
+    algorithm: RSA
+    size: 2048
+  secretName: bookinfo-cert
+EOF
+```
+### Deploy application
+we use bookinfo application to show an end to end encryption
+
+```bash
+oc new-project bookinfo
+oc apply -f ./app/smmr.yaml
+oc apply -f ./app/bookinfo.yaml
+oc apply -f ./app/destination-rule.yaml
+oc apply -f ./app/gateway.yaml
+```
+
+### Check SSL termination
+#### Check termination at ingress/node 
+ get the nodes' IP address and istio-ingress service ports
+
+ **Note:** If you don't get a response, check the security group and be sure that port 3000-32767 is allowed 
+
+```bash
+export WORKER_IP_ADDRESS=$(oc get nodes -l node-role.kubernetes.io/worker -o jsonpath='{range .items[*]}{.status.addresses[?(@.type=="InternalIP")].address}{"\n"}{end}' | head -n 1)
+echo "Worker IP Address $WORKER_IP_ADDRESS"
+export INGRESS_NODE_PORT=$(oc get svc -n ossm istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="https")].nodePort}')
+echo "Ingress service NodePort $INGRESS_NODE_PORT"
+```
+
+Check application endpoint
+```bash
+curl -v -k --resolve securedbookinfo.com:$INGRESS_NODE_PORT:$WORKER_IP_ADDRESS  https://securedbookinfo.com:$INGRESS_NODE_PORT/productpage 
+```
+
+#### Check TLS termination at the NLB layer
+
+
+ ```bash
+export NLB_URL=$(oc get svc -n ossm istio-ingressgateway -ojsonpath='{.status.loadBalancer.ingress[0].hostname}')
+echo "NLB's FQDN:  $NLB_URL"
+export NLB_IP=$(dig +short $NLB_URL | head -1)
+echo "NLP IP address $NLB_IP"
+```
+Check application  endpoint
```

Comment:
Generally speaking for readability you should leave a carriage return before and after all code blocks.
<empty line>
```bash
code
```
<empty line>
a sentence.